DOCS-4130: Add how to access resources from modules (#4453)
viamrobotics/docs: docs/operate/get-started/other-hardware/create-module/_index.md

3. **Decide on configuration attributes and dependencies**

   Make a list of required and optional attributes for users to configure when adding your module to a machine.
   Some examples of attributes:

   - A filepath from which to access data
   - A pin to which a device is wired
   - An optional signal frequency to override a default value

   You can also add dependencies to other resources that your module needs to use.

>    For example, if your module needs to access a camera, code your module to get a camera resource as a dependency.
>    This will allow you to access the camera with the camera API methods from within your module.
> 
>    You'll add these attributes and dependencies to the `Validate` and `Reconfigure` functions when you write the module.

Reviewer: I think the "code your module to get a resource as a dependency is not terribly helpful because we don't know how yet.
```suggestion
   For example, if your module needs to access a camera, add any required attributes and dependencies to the `Validate` and `Reconfigure` functions when you write the module.
   Then you can use the camera API methods from  within the module.
```

Author: I think that sounds as though you just add it in the config though

Reviewer: Is that not what you do?

Author: I mean, this page is focused on writing the module, not on configuring it. And if the module you're writing needs access to a camera, right now the important thing that you as a reader need to focus on coding the module's validate and reconfigure methods to get the camera as a dependency. The configuration part is a much later step, and configuring something doesn't make it a dependency.
